### PR TITLE
Feature/support coco tasks format

### LINKED ISSUE #1: Add all tasks for COCO dataset formats
Need for support for all **COCO** dataset formats that will be:

- [ x ] COCO Detection
- [ x ] COCO Segmentation
- [ x ] COCO Panoptic 
- [   ] COCO Keypoint detection
- [   ] COCO Image captioning
- [   ] COCO Dense pose
 
**🚨 IMPORTANT:** 
Follow the abstract class convention.

### PR DESCRIPTION
**Pull request by german36-del**
## Summary
This Pull Request include the following tasks:
- Keypoint detection
- Captioning
- Dense Pose
It also include a slight modification in the pull request template, the inclusion of show method inside the method __getittem__ and the modification of the way to load the images in each class for the respective task (Before a DataLoader from Pytorch was used but at the moment is not necessary, if in future releases we include features that need it it is straightforward to be changed)

Also included a function for ensuring that the output folder exists.
## Description
- Keypoint detection:
The class related is _COCODatasetKeypoint_ and the registered task _keypoint_.

Keypoint_example

Here is an example of the visualization. 
**NOTE**: I also scale and reshape the other fields, segmentation, area, bbox... I think it is the most logic strategy thinking in the final user.

- Captioning:
The class related is _COCODatasetCaption_ and the registered task _caption_.
It is opened one window per caption, sometimes one image can have several captions. The title of the image is as follows:
Caption: "A woman in the supermarket"

Caption_example

- Dense Pose:
The class related is _COCODatasetDensePose_ and the registered task _densepose_.

Basically the visualization is related to the Chart Indexes (I) and U, V.

Densepose_example

## How to test

For the Captioning and Keypoint download the annotations and the images in the official COCO page https://cocodataset.org/#home

And in the DensePose you can download here the annotations https://github.com/facebookresearch/detectron2/blob/main/projects/DensePose/doc/DENSEPOSE_DATASETS.md
It is only used the chart based annotations. In case of wanting to include also Continuous Surface Embeddings Annotations please answer to this PR.

### CODE REVIEW

File: .gitignore
Comment (javierganan99) [inline .gitignore:1]: Why *_build removed??

Comment (german36-del) [reply .gitignore:1]: No idea, must be an error when merging from main. Changing it asap.

Comment (fjganan14) [reply .gitignore:1]: I will add it again, no prob

**Comment by javierganan99** [comment]
I have reviewed all the code, and made some changes, specially for the visualization. I used opencv library for all the visualizations, removing the matplotlib dependency. At the end, the visualization for each task is:

- Detection:
detection
- Segmentation:
segmentation
- Panoptic:
panoptic
- Keypoints:
keypoints
keypoints_2
- Captions (Displayed one after the other for all the captions):
captions_1
captions_2
- Dense Pose:
densepose
